lstchain/conftest.py

```diff
@@ -15,11 +20,97 @@ def pytest_configure(config):
         if config.option.markexpr:
             config.option.markexpr += ' and '
         else:
-            config.option.markexpr +=  'not private_data'
-
+            config.option.markexpr += 'not private_data'
 
 
-@pytest.fixture(scope='session')
+@pytest.fixture(scope="session")
 def temp_dir():
+    """Shared temporal directory for the tests."""
+    with tempfile.TemporaryDirectory(prefix='test_lstchain') as d:
+        yield Path(d)
+
+
+@pytest.fixture(scope="session")
+def temp_dir_simulated_files():
+    """Temporal common directory for processing simulated data."""
+    with tempfile.TemporaryDirectory(prefix='test_lstchain') as d:
+        yield Path(d)
+
+
+@pytest.mark.private_data
+@pytest.fixture(scope="session")
+def temp_dir_observed_files():
+    """Temporal common directory for processing observed data."""
     with tempfile.TemporaryDirectory(prefix='test_lstchain') as d:
         yield Path(d)
+
+
+@pytest.fixture(scope="session")
+def mc_gamma_testfile():
+    """Get a simulated test file."""
+    return get_dataset_path('gamma_test_large.simtel.gz')
+
+
+@pytest.fixture(scope="session")
+def simulated_dl1_file(temp_dir_simulated_files, mc_gamma_testfile):
+    """Produce a dl1 file from simulated data."""
+    output_dl1_path = temp_dir_simulated_files / "dl1_gamma_test_large.h5"
+    run_program(
+        "lstchain_mc_r0_to_dl1",
+        "-f", mc_gamma_testfile,
+        "-o", temp_dir_simulated_files
+    )
+    return output_dl1_path
+
+
+@pytest.fixture(scope="session")
+def simulated_dl2_file(temp_dir_simulated_files, simulated_dl1_file, rf_models):
+    """
+    Produce the test dl2 file from the simulated dl1 test file
+    using the random forest test models.
+    """
+    dl2_file = temp_dir_simulated_files / "dl2_gamma_test_large.h5"
+    run_program(
+        "lstchain_dl1_to_dl2",
+        "--input-file",
+        simulated_dl1_file,
+        "--path-models",
+        rf_models["path"],
+        "--output-dir",
+        temp_dir_simulated_files
+    )
+    return dl2_file
+
+
+@pytest.fixture(scope="session")
+def fake_dl1_proton_file(temp_dir_simulated_files, simulated_dl1_file):
+    """
+    Produce a fake dl1 proton file by copying the dl2 gamma test file
+    and changing mc_type.
+    """
+    dl1_proton_file = temp_dir_simulated_files / 'dl1_fake_proton.simtel.h5'
+    events = pd.read_hdf(simulated_dl1_file, key=dl1_params_lstcam_key)
+    events.mc_type = 101
+    events.to_hdf(dl1_proton_file, key=dl1_params_lstcam_key)
+    return dl1_proton_file
+
+
+@pytest.fixture(scope="session")
+def rf_models(temp_dir_simulated_files, simulated_dl1_file):
+    """Produce test random forest models."""
+    gamma_file = simulated_dl1_file
+    proton_file = simulated_dl1_file
+    models_path = temp_dir_simulated_files
+    file_model_energy = models_path / "reg_energy.sav"
```

Comment:
These are pickled models, right? Why then not use the `.pickle` extension?

Reply:
Do you mean just modify the filenames to `file_model_energy = models_path / "reg_energy.pickle"` an so on?

Reply:
yes. But this is a really minor thing.

Reply:
Yes, sure. I can change them. Not sure I get the points of this `.pickle` extension. It is a file that you open, dump something in it or modify it and afterward, you can recover the original content for using it in another test for example?

Reply:
It is just about using the correct extension for the correct filetype. We also use `.h5` for the temporary hdf5 files ;)

Reply:
See: https://stackoverflow.com/a/40433504/3838691

Reply:
we are currently using `.sav` in several places in the repo:
https://github.com/cta-observatory/cta-lstchain/search?q=.sav

let's try to be consistent (either we change all or leave `.sav` here for the time being)

Reply:
I was going to say that. I've always seen `.sav` extension so I did not understand at the beginning.

Reply:
Ah ok, then let's not change this here and leave that for another time.